### PR TITLE
arch/sim: Enable SIM_WALLTIME option by default

### PR DESCRIPTION
**Pull request by sebastianene07**
### Summary of Changes ###

Running the NuttX simulation 'as fast as possible' breaks the features
that depend on timing: eg. the Bluetooth stack. Enabling this option by
default SIM_WALLTIME=y will introduce delays and will tick the simulation at
a real pace.

## Impact

We discussed here the potential impact #2084 and we agreed to enable this option by default.

## Testing

Signed-off-by: Sebastian Ene <sene@apache.org>

### CODE REVIEW

**Comment by btashton** [comment]
You are going to need to refresh these configs:
```
sim/sim/sim/configs/minibasic/defconfig:CONFIG_SIM_WALLTIME=y
sim/sim/sim/configs/unionfs/defconfig:CONFIG_SIM_WALLTIME=y
sim/sim/sim/configs/smp/defconfig:CONFIG_SIM_WALLTIME=y
sim/sim/sim/configs/loadable/defconfig:CONFIG_SIM_WALLTIME=y
sim/sim/sim/configs/bthcisock/defconfig:CONFIG_SIM_WALLTIME=y
sim/sim/sim/configs/tcploop/defconfig:CONFIG_SIM_WALLTIME=y
sim/sim/sim/configs/rpproxy/defconfig:CONFIG_SIM_WALLTIME=y
sim/sim/sim/configs/duktape/defconfig:CONFIG_SIM_WALLTIME=y
sim/sim/sim/configs/ipforward/defconfig:CONFIG_SIM_WALLTIME=y
sim/sim/sim/configs/nshcromfs/defconfig:CONFIG_SIM_WALLTIME=y
sim/sim/sim/configs/nsh/defconfig:CONFIG_SIM_WALLTIME=y
sim/sim/sim/configs/bas/defconfig:CONFIG_SIM_WALLTIME=y
```

**Comment by sebastianene07** [comment]
> You are going to need to refresh these configs:
> 
> ```
> sim/sim/sim/configs/minibasic/defconfig:CONFIG_SIM_WALLTIME=y
> sim/sim/sim/configs/unionfs/defconfig:CONFIG_SIM_WALLTIME=y
> sim/sim/sim/configs/smp/defconfig:CONFIG_SIM_WALLTIME=y
> sim/sim/sim/configs/loadable/defconfig:CONFIG_SIM_WALLTIME=y
> sim/sim/sim/configs/bthcisock/defconfig:CONFIG_SIM_WALLTIME=y
> sim/sim/sim/configs/tcploop/defconfig:CONFIG_SIM_WALLTIME=y
> sim/sim/sim/configs/rpproxy/defconfig:CONFIG_SIM_WALLTIME=y
> sim/sim/sim/configs/duktape/defconfig:CONFIG_SIM_WALLTIME=y
> sim/sim/sim/configs/ipforward/defconfig:CONFIG_SIM_WALLTIME=y
> sim/sim/sim/configs/nshcromfs/defconfig:CONFIG_SIM_WALLTIME=y
> sim/sim/sim/configs/nsh/defconfig:CONFIG_SIM_WALLTIME=y
> sim/sim/sim/configs/bas/defconfig:CONFIG_SIM_WALLTIME=y
> ```

I refreshed these configs.

**Review by btashton — APPROVED** [review]

File: arch/sim/Kconfig [hunk 1]
```diff
@@ -99,7 +99,7 @@ endchoice
 
 config SIM_WALLTIME
 	bool "Run the simulation at a fixed cadence in near real-time"
-	default n
+	default y
```
[the remark this answers is not in the document]
Comment (sebastianene07) [reply arch/sim/Kconfig:102]: I will edit the Kconfig and add a depends on rule for SIM_HCISOCKET.

[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply arch/sim/Kconfig:102]: But other place use select, should we unify the usage?

[the remark this answers is not in the document]
Comment (v01d) [reply arch/sim/Kconfig:102]: If the default is `y`, I think it is OK to have it depend on it. When a default is `n`, the depend makes it hard to discover the option, in that case `select` is more useful.

**Review by v01d — APPROVED** [review]

File: arch/sim/Kconfig (hunk 1 above)
[the remark this answers is not in the document]
Comment (v01d) [reply arch/sim/Kconfig:102]: If you are OK with this @xiaoxiang781216 please merge.

[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply arch/sim/Kconfig:102]: @v01d Either select or depend is fine for me, but we need ensure the usage consistence: it's very strange that SIM_HCISOCKET depend on SIM_WALLTIME but SIM_NETDEV and SIM_X11FB select SIM_WALLTIME in the same Kconfig.

[the remark this answers is not in the document]
Comment (v01d) [reply arch/sim/Kconfig:102]: I'd imagine it was due to WALLTIME being `n` by default. In that case I agree that those should also be changed into `depends`

[the remark this answers is not in the document]
Comment (sebastianene07) [reply arch/sim/Kconfig:102]: I changed these options to depend on ```SIM_WALLTIME```

File: arch/sim/Kconfig [hunk 2]
```diff
@@ -134,10 +134,7 @@ endif
 config SIM_NETDEV
 	bool "Simulated Network Device"
 	default y
-	depends on NET_ETHERNET
-	select ARCH_HAVE_NETDEV_STATISTICS
-	select SCHED_LPWORK
-	select SIM_WALLTIME
+	depends on (NET_ETHERNET && SIM_WALLTIME && ARCH_HAVE_NETDEV_STATISTICS && SCHED_LPWORK)
```
Comment (xiaoxiang781216) [inline arch/sim/Kconfig:137]: ARCH_HAVE_NETDEV_STATISTICS is designed to be selected.

Comment (xiaoxiang781216) [reply arch/sim/Kconfig:137]: and the most place select SCHED_LPWORK not depend on SCHED_LPWORK

Comment (xiaoxiang781216) [inline arch/sim/Kconfig:137]: ARCH_HAVE_NETDEV_STATISTICS is designed to be selected, and the most place select SCHED_LPWORK, not depend on SCHED_LPWORK. so it's better to change:
	depends on NET_ETHERNET && SIM_WALLTIME
	select ARCH_HAVE_NETDEV_STATISTICS
	select SCHED_LPWORK

Comment (sebastianene07) [reply arch/sim/Kconfig]: Ok I updated to this

File: arch/sim/Kconfig [hunk 3]
```diff
@@ -225,8 +224,7 @@ if SIM_FRAMEBUFFER
 config SIM_X11FB
 	bool "Use X11 window"
 	default n
-	select SCHED_LPWORK
-	select SIM_WALLTIME
+	depends on (SCHED_LPWORK && SIM_WALLTIME)
```
Comment (xiaoxiang781216) [inline arch/sim/Kconfig:227]: need keep SCHED_LPWORK as select too

Comment (sebastianene07) [reply arch/sim/Kconfig:227]: oh, ok

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM.

**Pull request merged**